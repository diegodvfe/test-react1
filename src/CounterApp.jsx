import { useState } from 'react';
import PropTypes from 'prop-types';



export const CounterApp = ({ value }) => {
    const [counter, setCounter] = useState(value)
    
    
const handleAdd = () => {
    // console.log(event);
    setCounter( (c) => c +1 )

    
};

const handleLess = () => {
   setCounter((c) => c -1)

}

const handleReset = () => {
    setCounter ( () =>value )
}
console.log(handleReset)


    return (
        <>
            <h1>Counter App</h1>
            <h2> {counter} </h2>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleLess}>-1</button>
            <button onClick={handleReset} >Resest</button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number
};

CounterApp.defaultProps = {
    value: 'No hay valor'
};
