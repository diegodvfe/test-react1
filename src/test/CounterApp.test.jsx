
import { CounterApp } from './../CounterApp';
import { fireEvent, render, screen } from '@testing-library/react';

describe('Pruebas del component CounterApp', () => {
  
      const intialValue = 10;

      // 1
      test('Debe de hacer match con el snapshot', () => {

            const { container } = render(<CounterApp value={intialValue} />)
            expect (container).toMatchSnapshot();
        
      })

      test('Debe de mostrar e valor inicial de 100 en <CounterApp value={100} />', () => {
            
      render(<CounterApp value={100} />);
      expect(screen.getByText(100)).toBeTruthy();


      })

      test('debe de incrementar con el boton +1', () => {
        render( <CounterApp value={intialValue} />)

        fireEvent.click( screen.getByText('+1'))
        expect( screen.getByText('11')).toBeTruthy();
        
      })
      
      test('debe de decrementar con el boton -1 ', () => {
            
            render( <CounterApp value={intialValue}/>)
            fireEvent.click( screen.getByText('-1'))
             expect( screen.getByText('9')).toBeTruthy()
             screen.debug()
      })

      test('debe de funcionar el boton de reset', () => {
        render ( <CounterApp value={ 355 } />)

        fireEvent.click( screen.getByText('+1'));
        fireEvent.click( screen.getByText('Resest'));

        expect( screen.getByText(355)).toBeTruthy();



      })
      
      
      
      
      

})
