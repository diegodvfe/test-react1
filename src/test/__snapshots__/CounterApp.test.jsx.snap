// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del component CounterApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
     
    10
     
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Resest
  </button>
</div>
`;
